File: src/hm-get-mode.pl
#!/usr/bin/env perl

use strict;
use warnings;

if (@ARGV != 1) {
	print STDERR "usage: $0 FILENAME\n";
	exit 0;
}

my @s = lstat $ARGV[0] or die $!;
printf "%06o\n", $s[2];

File: src/hm-get-mode.c
#include <stdio.h>
#include <stdlib.h>
#include <sys/stat.h>

int main(int argc, char* argv[])
{
	if (argc != 2)
	{
		fprintf(stderr, "usage: %s FILENAME\n", argv[0]);
		return EXIT_FAILURE;
	}

	struct stat s;
	if (lstat(argv[1], &s) != 0) {
		perror("lstat");
		return EXIT_FAILURE;
	}

	printf("%06o\n", s.st_mode);

	return EXIT_SUCCESS;
}
